biz.aQute.bndlib.tests/test/test/annotationheaders/spi/consumerC/SPIServiceFactory.java

```diff
@@ -0,0 +1,13 @@
+package test.annotationheaders.spi.consumerC;
+
+import aQute.bnd.annotation.spi.ServiceConsumer;
+import aQute.bnd.annotation.spi.ServiceConsumer.Resolution;
+import test.annotationheaders.spi.SPIService;
+
+public class SPIServiceFactory {
+
+	@ServiceConsumer(value = SPIService.class, resolution = Resolution.OPTIONAL)
```

Comment:
yeah, sounds reasonable since I took the less complex approach initially of not doing bytecode analysis I think it's best to limit to type & package as suggested so people don't read more into it than necessary.